index.js

```diff
@@ -343,42 +362,55 @@ function processStyle(glStyle, map, baseUrl, host, path, accessToken) {
                     tileGrid.getResolution(tileGrid.getMinZoom()));
                 }
                 unByKey(key);
+              } else if (tilejson.getState() == 'error') {
+                layer.setSource(undefined);
+                unByKey(key);
               }
             });
             return layer;
           })();
         } else if (glSource.type == 'raster') {
-          let source;
-          if (!glSource.tiles) {
-            source = new TileJSON({
-              transition: transition,
-              url: url,
-              crossOrigin: 'anonymous'
-            });
-          } else {
-            source = new XYZ({
-              transition: transition,
-              attributions: glSource.attribution,
-              minZoom: glSource.minzoom,
-              maxZoom: 'maxzoom' in glSource ? glSource.maxzoom : 22,
-              tileSize: glSource.tileSize || 512,
-              url: url,
-              urls: glSource.tiles,
-              crossOrigin: 'anonymous'
+          layer = (function() {
+            const layer = new TileLayer({
+              visible: glLayer.layout ? glLayer.layout.visibility !== 'none' : true
             });
-            transition = 0;
-          }
-          source.setTileLoadFunction(function(tile, src) {
-            if (src.indexOf('{bbox-epsg-3857}') != -1) {
-              const bbox = source.getTileGrid().getTileCoordExtent(tile.getTileCoord());
-              src = src.replace('{bbox-epsg-3857}', bbox.toString());
+            let source = undefined;
+            if (!glSource.tiles) {
+              source = new TileJSON({
+                transition: transition,
+                url: url,
+                crossOrigin: 'anonymous'
+              });
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define this function outside of a loop. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)